---
title: "RNA-seq"
author: "Sofia Salazar & Bruno Villalobos"
date: "3/9/2022"
output: html_document
---

## Introduction

In this report we are going to do an analisis of RNA-seq data using the following pipeline, starting with the raw read RNA-seq data from the experiment SRP127520 **"Knockdown of FOXP1 promotes the development of lung adenocarcinoma"**.

## Pipeline

### Quality control

The first step in our pipeline for the analysis is to make a quality report of the raw reads we are provided in order to assess possible quality control strategies and therefore be able to make good conclusion from the full analysis.

For the quality report we are going to use the Multiqc module, using the fastq files stored in the following path

```{bash eval = F}
/mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520
```
 
We run the following script from our directory

```{bash}
#!/bin/bash
# Use current working directory
#$ -cwd
#
# Join stdout and stderr
#$ -j y
#
# Run job through bash shell
#$ -S /bin/bash
#
#You can edit the script since this line
#
# Your job name
#$ -N MULTIQC
#
# Send an email after the job has finished
#$ -m e
#$ -M sogusama02@gmail.com
#
#
# If modules are needed, source modules environment (Do not delete the next line):
. /etc/profile.d/modules.sh
#
# Add any modules you might require:
module load fastqc/0.11.3
module load multiqc/1.5
#
# Write your commands in the next line
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417885.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417885.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417886.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417886.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417887.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417887.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417888.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417888.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417889.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417889.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417890.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417890.sra_2.fastq.gz -o .
multiqc .
```


```{bash eval = F}
cd /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC
qsub multiqc1.sge
```

And downloading the multiqc report
```{bash}
rsync -rptuvl ssalazar@dna.lavis.unam.mx:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC/multiqc_report1.html .
```

Based on the report, we are discarding sequences with less than 30Q and also trimmin the ILLUMINA adapter sequences.

## Trimming

For the trimming, we are using the trimmomatic tool, using the following arguments

**PE:** Because we are using paired end sequences

**-phred33:** We are specifing the base quality encoding

**ILLUMINACLIP:** We use the adapters in the fasta file, :2:30:10 are mismatch/accuracy treshold for adapter/reads pairing.

**SLIDINGWINDOW:5:30** Scaning the reads in a 4 base window. If mean quality under 15 in that window, the read is trimmed in the leftmost position of the window.

**MINLEN:** is the minimum length of reads we are accepting, if they are shorted after trimming, we discard them.

The full script, adding the command to create a multiqc report of the trimmed sequences is the following:

```{bash}
#!/bin/bash
# Use current working directory
#$ -cwd
#
# Join stdout and stderr
#$ -j y
#
# Run job through bash shell
#$ -S /bin/bash
#
#You can edit the script since this line
#
# Your job name
#$ -N PEtrimming
#
# Send an email after the job has finished
#$ -m e
#$ -M sogusama02@gmail.com
#
#
# If modules are needed, source modules environment (Do not delete the next line):
. /etc/profile.d/modules.sh
#
# Add any modules you might require:
module load trimmomatic/0.33
module load fastqc/0.11.3
module load multiqc/1.5
#
# Write your commands in the next line
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417885.sra_1.fastq.gz -baseout ./SRR6417885_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417886.sra_1.fastq.gz -baseout ./SRR6417886_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417887.sra_1.fastq.gz -baseout ./SRR6417887_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417888.sra_1.fastq.gz -baseout ./SRR6417888_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417889.sra_1.fastq.gz -baseout ./SRR6417889_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417890.sra_1.fastq.gz -baseout ./SRR6417890_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
fastqc ./SRR6417885* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417886* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417887* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417888* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417889* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417890* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
multiqc /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
```

```{bash}
qsub trimming-qc.sge
```
And downloading the multiqc report
```{bash}
rsync -rptuvl ssalazar@dna.lavis.unam.mx:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2/multiqc_report2.html .
```

### Alignment

We are using transcriptome mapping because:
we have paired end??
human transcriptome is already annotated 
we are mainly interested in transcriptome

We first download the reference transcriptome from GENCODE
```{bash}
# from rna-seq/tarea
mkdir resources
cd resources
wget http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/gencode.v38.transcripts.fa.gz
```

Making the transcriptome index

```{bash}
module load kallisto/0.45.0
kallisto index -i index_kallisto45_gencode_human_v38 gencode.v38.transcripts.fa.gz
```

### Quantification

We first make directories for each sample
```{bash}
cd /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out
mkdir kallisto 
ls ../trimmed/ | perl -pe 's/(SRR\d+)_trmd.*/mkdir $1/' | uniq
```


```{bash}
mkdir SRR6417885
mkdir SRR6417886
mkdir SRR6417887
mkdir SRR6417888
mkdir SRR6417889
mkdir SRR6417890
```

generate lines for our bash script

```{bash}
ls | grep "SRR" | perl -pe 's/(SRR\d+)/kallisto quant -i ..\/..\/resources\/index_kallisto45_gencode_human_v38 -o .\/$1 ..\/trimmed\/$1* /'
```

Bash script
```{bash}
#!/bin/bash
# Use current working directory
#$ -cwd
#
# Join stdout and stderr
#$ -j y
#
# Run job through bash shell
#$ -S /bin/bash
#
#You can edit the scriptsince this line
#
# Your job name
#$ -N Quantification
#
# Send an email after the job has finished
#$ -m e
#$ -M sogusama02@gmail.com
#
#
# If modules are needed, source modules environment (Do not delete the next line):
. /etc/profile.d/modules.sh
#
# Add any modules you might require:
module load kallisto/0.45.0
#
# Write your commands in the next line
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417885 ../trimmed/SRR6417885* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417886 ../trimmed/SRR6417886* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417887 ../trimmed/SRR6417887* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417888 ../trimmed/SRR6417888* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417889 ../trimmed/SRR6417889* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417890 ../trimmed/SRR6417890* 
```

```{bash}
qsub quantification.sge
```

### Data integration

Using tximport

```{bash}
screen -S txi
cd /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/kallisto
module load r/4.0.2
R
```


```{r}
library(tximport)
library(tidyverse)
files <- file.path("/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/kallisto", list.dirs("."), "abundance.h5")
names(files) <- str_extract(files,"SRR\\d+")
files <- files[c(2, 3, 4, 5, 6, 7)]
```

Transcripts need to be associated with gene IDs for gene-level summarization
For kallisto, the files only provide the transcript ID.
We first make a data.frame called tx2gene with two columns: 1) transcript ID and 2) gene ID.

```{r}
# Loading tables with the info

# transcript id - gene id correspondence
tx2gene <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.v38.basic.pc.transcripts.enstid_ensgid-nover.csv",stringsAsFactors = F)
# transcript id - gene name correspondence
tx2genename <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.v38.basic.pc.transcripts.enstid_genename-nover.csv",stringsAsFactors = F)

```


Runing tximport to summarize transcript-level abundance estimates for transcript- and gene-level analysis.

```{r}
txi.kallisto <- tximport(files, type = "kallisto", tx2gene = tx2gene, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)
txi.kallisto.name <- tximport(files, type = "kallisto", tx2gene = tx2genename, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)

```

```{r}
names(txi.kallisto)
```

[1] "abundance"           "counts"              "length"             
[4] "countsFromAbundance"

```{r}
head(txi.kallisto$counts)
```

                 SRR6417885  SRR6417886 SRR6417887 SRR6417888 SRR6417889
ENSG000000000035  48.856706 36.12039819  24.279581  39.413793  37.693556
ENSG00000000005    0.000000  0.00000000   0.000000   0.000000   0.000000
ENSG000000004194   0.000000  1.81937909   0.000000   5.729418  22.804753
ENSG000000004574   6.663615  0.01752127   7.742207   3.390685   8.781861
ENSG000000004607   9.369975 11.00000000  13.829627   8.001456  16.942265
ENSG000000009383   0.000000  0.00000000   0.000000   0.000000   0.000000
                 SRR6417890
ENSG000000000035  46.855202
ENSG00000000005    0.000000
ENSG000000004194  23.005803
ENSG000000004574   0.000000
ENSG000000004607   6.367758
ENSG000000009383   0.000000

### Normalization

```{r}
samples <- read.csv("/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/resources/samples.csv",stringsAsFactors = F, header = TRUE) # obtained parsing file on SRA Run Selector 
samples <- column_to_rownames(samples, var = "Run")
samples
```

      Assay.Type AvgSpotLen       Bases  BioProject    BioSample
SRR6417885    RNA-Seq        300 12773176500 PRJNA427471 SAMN08242902
SRR6417886    RNA-Seq        300 11734114800 PRJNA427471 SAMN08242901
SRR6417887    RNA-Seq        300 12898263600 PRJNA427471 SAMN08242900
SRR6417888    RNA-Seq        300 11938590300 PRJNA427471 SAMN08242899
SRR6417889    RNA-Seq        300 14398560300 PRJNA427471 SAMN08242898
SRR6417890    RNA-Seq        300 13948899900 PRJNA427471 SAMN08242897
                Bytes Cell_Line Center.Name Consent DATASTORE.filetype
SRR6417885 5233611886      A549         GEO  public          fastq,sra
SRR6417886 4896761293      A549         GEO  public          fastq,sra
SRR6417887 5304248538      A549         GEO  public          fastq,sra
SRR6417888 5062237114      A549         GEO  public          fastq,sra
SRR6417889 5962982366      A549         GEO  public          fastq,sra
SRR6417890 5745034790      A549         GEO  public          fastq,sra
           DATASTORE.provider               DATASTORE.region Experiment
SRR6417885         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 SRX3510910
SRR6417886         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 SRX3510911
SRR6417887         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 SRX3510912
SRR6417888         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 SRX3510913
SRR6417889         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 SRX3510914
SRR6417890         gs,ncbi,s3 gs.US,ncbi.public,s3.us-east-1 SRX3510915
                  Genotype GEO_Accession..exp.          Instrument
SRR6417885       wild type          GSM2902357 Illumina HiSeq 2000
SRR6417886       wild type          GSM2902358 Illumina HiSeq 2000
SRR6417887       wild type          GSM2902359 Illumina HiSeq 2000
SRR6417888 FOXP1 knockdown          GSM2902360 Illumina HiSeq 2000
SRR6417889 FOXP1 knockdown          GSM2902361 Illumina HiSeq 2000
SRR6417890 FOXP1 knockdown          GSM2902362 Illumina HiSeq 2000
           LibraryLayout LibrarySelection  LibrarySource     Organism Platform
SRR6417885        PAIRED             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA
SRR6417886        PAIRED             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA
SRR6417887        PAIRED             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA
SRR6417888        PAIRED             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA
SRR6417889        PAIRED             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA
SRR6417890        PAIRED             cDNA TRANSCRIPTOMIC Homo sapiens ILLUMINA
                    ReleaseDate Sample.Name                      source_name
SRR6417885 2018-02-11T00:00:00Z  GSM2902357          A549 cell lines;control
SRR6417886 2018-02-11T00:00:00Z  GSM2902358          A549 cell lines;control
SRR6417887 2018-02-11T00:00:00Z  GSM2902359          A549 cell lines;control
SRR6417888 2018-02-11T00:00:00Z  GSM2902360  FOXP1 knockdown A549 cell lines
SRR6417889 2018-02-11T00:00:00Z  GSM2902361 FOXP1 knockdown A549 cell linesl
SRR6417890 2018-02-11T00:00:00Z  GSM2902362  FOXP1 knockdown A549 cell lines
           SRA.Study              tissue
SRR6417885 SRP127520 lung adenocarcinoma
SRR6417886 SRP127520 lung adenocarcinoma
SRR6417887 SRP127520 lung adenocarcinoma
SRR6417888 SRP127520 lung adenocarcinoma
SRR6417889 SRP127520 lung adenocarcinoma
SRR6417890 SRP127520 lung adenocarcinoma

```{r}
library(DESeq2)
dds <- DESeqDataSetFromTximport(txi.kallisto,
                                   colData = samples,
                                   design = ~ Genotype) 
# + si hay vars en condiciones
```

```{r}
dds
```

class: DESeqDataSet
dim: 19453 6
metadata(1): version
assays(2): counts avgTxLength
rownames(19453): ENSG000000000035 ENSG00000000005 ... ENSG00000288722
ENSG00000288725
rowData names(0):
colnames(6): SRR6417885 SRR6417886 ... SRR6417889 SRR6417890
colData names(26): Assay.Type AvgSpotLen ... SRA.Study tissue

```{r}
# Prefiltrado para quitar genes con cuentas bajas
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep, ]
dds$Genotype <- factor(dds$Genotype, levels = c("wild type","FOXP1 knockdown"))
dds <- DESeq(dds)
```

```{r}
# use contrast= parameter if design formula has > 2 levels
res <- results(dds)
res
```

log2 fold change (MLE): Genotype FOXP1.knockdown vs wild.type
Wald test p-value: Genotype FOXP1.knockdown vs wild.type
DataFrame with 10831 rows and 6 columns
                  baseMean log2FoldChange     lfcSE      stat     pvalue
                 <numeric>      <numeric> <numeric> <numeric>  <numeric>
ENSG000000000035  40.52100     -0.0524047  0.434333 -0.120656 0.90396381
ENSG000000004194   8.02003      4.2945438  1.322891  3.246332 0.00116903
ENSG000000004574   4.41804     -0.6385878  1.566245 -0.407719 0.68347991
ENSG000000004607  11.33065     -0.5514932  0.682694 -0.807819 0.41919494
ENSG000000010843  30.27042     -0.1124562  0.384476 -0.292492 0.76991031
...                    ...            ...       ...       ...        ...
ENSG00000288701  304.67259       0.188924  0.100165  1.886114  0.0592795
ENSG00000288702   11.83816       0.278061  0.985937  0.282027  0.7779225
ENSG00000288709   30.72393      -0.291760  0.359618 -0.811305  0.4171906
ENSG00000288710    2.76967      -0.363605  2.146279 -0.169412  0.8654729
ENSG00000288722   24.39237      -0.410888  1.396288 -0.294271  0.7685505
                      padj
                 <numeric>
ENSG000000000035  0.981802
ENSG000000004194        NA
ENSG000000004574        NA
ENSG000000004607        NA
ENSG000000010843  0.941239
...                    ...
ENSG00000288701   0.316297
ENSG00000288702         NA
ENSG00000288709   0.788419
ENSG00000288710         NA
ENSG00000288722         NA

## Volcano Plot

In order to visualize the normalized data we can make use, in this case, of a volcano plot. In this way, we will be able to easily account for the genes that have a higher p-value significance.
The arguments wont vary from what was given in class, except for the out directroy.


```{r}
library(ggplot2) # we load ggplot2
outdir = "/mnt/Timina/bioinfoII/bvillalobos/rna-seq/results/"
png(file = paste0(outdir,"volcano01-res.png"),
    width = 800, height = 800) # we will save as a png
# The basic scatter plot: x is "log2FoldChange", y is "pvalue"
ggplot(data=as.data.frame(res), aes(x=log2FoldChange, y=pvalue)) +
  geom_point() # scatter plot
dev.off()
```

We then make a more clear version of the plot:

```{r}
png(file = paste0(outdir,"volcano02-res.png"),
    width = 800, height = 800) # save as a png
ggplot(data=as.data.frame(res), aes(x=log2FoldChange, y=-log10(pvalue))) +
  geom_point() # scatter plot
dev.off()
```

Then we mark a threshold for significance.

```{r}
png(file = paste0(outdir,"volcano03-res.png"),
    width = 800, height = 800) # save as png
ggplot(data=as.data.frame(res), aes(x=log2FoldChange, y=-log10(pvalue))) +
  geom_point() + # scatter plot
  theme_minimal() +
  geom_vline(xintercept=c(-0.6, 0.6), col="red") + # vertical lines for log2FoldChange thresholds
  geom_hline(yintercept=-log10(0.05), col="red") + # horizontal line for the p-value threshold
  xlim(-15,15)
dev.off()
```

We can classify our data by the given threshold in down, up and no differentially expressed. 

```{r}
# Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2FoldChange respectively positive or negative)
de <- as.data.frame(res)
# add a column of NAs
de$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
de$diffexpressed[de$log2FoldChange > 0.6 & de$pvalue < 0.05] <- "UP"
# if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"
de$diffexpressed[de$log2FoldChange < -0.6 & de$pvalue < 0.05] <- "DOWN"
png(file = paste0(outdir,"volcano04-res.png"),
    width = 800, height = 800) # save as png
ggplot(data=de, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed)) + 
  geom_point() + theme_minimal() + 
  geom_vline(xintercept=c(-0.6, 0.6), col="red") +
  geom_hline(yintercept=-log10(0.05), col="red") +
  xlim(-15,15)
dev.off()
```

Finally, we can add the names of the interest genes, which in this case its unclear whether they should be or not.

```{r}
library(ggrepel) # this avoids interlap betewwn text
# Create a new column "names" to de, that will contain the name of a subset if genes differentially expressed (NA in case they are not)
de$names <- NA
# filter for a subset of interesting genes
filter <- which(de$diffexpressed != "NO" & de$padj < 0.05 & (de$log2FoldChange >= 9 | de$log2FoldChange <= -9))
de$names[filter] <- rownames(de)[filter]
# Create and  save plot
png(file = paste0(outdir,"volcano05-res.png"),
    width = 800, height = 800) # save as png
ggplot(data=de, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=names)) +
    geom_point() +
    scale_color_manual(values=c("blue", "black", "red")) + # change data color
    theme_minimal() +
    geom_text_repel() +
    xlim(-15,15)
dev.off()
```


## References

https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html

http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf

http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/latest_release/_README.TXT

https://pachterlab.github.io/kallisto/manual

