---
title: "RNA-seq"
author: "Sofia Salazar & Bruno Villalobos"
date: "3/9/2022"
output: html_document
---

## Introduction

In this report we are going to do an analisis of RNA-seq data using the following pipeline, starting with the raw read RNA-seq data from the experiment SRP127520 **"Knockdown of FOXP1 promotes the development of lung adenocarcinoma"**.

## Pipeline

### Quality control

The first step in our pipeline for the analysis is to make a quality report of the raw reads we are provided in order to assess possible quality control strategies and therefore be able to make good conclusion from the full analysis.

For the quality report we are going to use the Multiqc module, using the fastq files stored in the following path

```{bash eval = F}
/mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520
```
 
We run the following script from our directory

```{bash}
#!/bin/bash
# Use current working directory
#$ -cwd
#
# Join stdout and stderr
#$ -j y
#
# Run job through bash shell
#$ -S /bin/bash
#
#You can edit the script since this line
#
# Your job name
#$ -N MULTIQC
#
# Send an email after the job has finished
#$ -m e
#$ -M sogusama02@gmail.com
#
#
# If modules are needed, source modules environment (Do not delete the next line):
. /etc/profile.d/modules.sh
#
# Add any modules you might require:
module load fastqc/0.11.3
module load multiqc/1.5
#
# Write your commands in the next line
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417885.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417885.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417886.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417886.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417887.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417887.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417888.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417888.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417889.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417889.sra_2.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417890.sra_1.fastq.gz -o .
fastqc /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417890.sra_2.fastq.gz -o .
multiqc .
```


```{bash eval = F}
cd /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC
qsub multiqc1.sge
```

And downloading the multiqc report
```{bash}
rsync -rptuvl ssalazar@dna.lavis.unam.mx:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC/multiqc_report1.html .
```

Based on the report, we are discarding sequences with less than 30Q and also trimmin the ILLUMINA adapter sequences.

## Trimming

For the trimming, we are using the trimmomatic tool, using the following arguments

**PE:** Because we are using paired end sequences

**-phred33:** We are specifing the base quality encoding

**ILLUMINACLIP:** We use the adapters in the fasta file, :2:30:10 are mismatch/accuracy treshold for adapter/reads pairing.

**SLIDINGWINDOW:5:30** Scaning the reads in a 4 base window. If mean quality under 15 in that window, the read is trimmed in the leftmost position of the window.

**MINLEN:** is the minimum length of reads we are accepting, if they are shorted after trimming, we discard them.

The full script, adding the command to create a multiqc report of the trimmed sequences is the following:

```{bash}
#!/bin/bash
# Use current working directory
#$ -cwd
#
# Join stdout and stderr
#$ -j y
#
# Run job through bash shell
#$ -S /bin/bash
#
#You can edit the script since this line
#
# Your job name
#$ -N PEtrimming
#
# Send an email after the job has finished
#$ -m e
#$ -M sogusama02@gmail.com
#
#
# If modules are needed, source modules environment (Do not delete the next line):
. /etc/profile.d/modules.sh
#
# Add any modules you might require:
module load trimmomatic/0.33
module load fastqc/0.11.3
module load multiqc/1.5
#
# Write your commands in the next line
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417885.sra_1.fastq.gz -baseout ./SRR6417885_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417886.sra_1.fastq.gz -baseout ./SRR6417886_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417887.sra_1.fastq.gz -baseout ./SRR6417887_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417888.sra_1.fastq.gz -baseout ./SRR6417888_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417889.sra_1.fastq.gz -baseout ./SRR6417889_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
trimmomatic PE -phred33 -basein /mnt/Timina/bioinfoII/rnaseq/tarea/data/SRP127520/SRR6417890.sra_1.fastq.gz -baseout ./SRR6417890_trmd.sra.fastq.gz ILLUMINACLIP:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/data/adapters/adapters-paired.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40
fastqc ./SRR6417885* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417886* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417887* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417888* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417889* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
fastqc ./SRR6417890* -o /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
multiqc /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2
```

```{bash}
qsub trimming-qc.sge
```
And downloading the multiqc report
```{bash}
rsync -rptuvl ssalazar@dna.lavis.unam.mx:/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/QC2/multiqc_report2.html .
```

### Alignment

We are using transcriptome mapping because:
we have paired end??
human transcriptome is already annotated 
we are mainly interested in transcriptome

We first download the reference transcriptome from GENCODE
```{bash}
# from rna-seq/tarea
mkdir resources
cd resources
wget http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/gencode.v38.transcripts.fa.gz
```

Making the transcriptome index

```{bash}
module load kallisto/0.45.0
kallisto index -i index_kallisto45_gencode_human_v38 gencode.v38.transcripts.fa.gz
```

### Quantification

We first make directories for each sample
```{bash}
cd /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out
mkdir kallisto 
ls ../trimmed/ | perl -pe 's/(SRR\d+)_trmd.*/mkdir $1/' | uniq
```


```{bash}
mkdir SRR6417885
mkdir SRR6417886
mkdir SRR6417887
mkdir SRR6417888
mkdir SRR6417889
mkdir SRR6417890
```

generate lines for our bash script

```{bash}
ls | grep "SRR" | perl -pe 's/(SRR\d+)/kallisto quant -i ..\/..\/resources\/index_kallisto45_gencode_human_v38 -o .\/$1 ..\/trimmed\/$1* /'
```

Bash script
```{bash}
#!/bin/bash
# Use current working directory
#$ -cwd
#
# Join stdout and stderr
#$ -j y
#
# Run job through bash shell
#$ -S /bin/bash
#
#You can edit the scriptsince this line
#
# Your job name
#$ -N Quantification
#
# Send an email after the job has finished
#$ -m e
#$ -M sogusama02@gmail.com
#
#
# If modules are needed, source modules environment (Do not delete the next line):
. /etc/profile.d/modules.sh
#
# Add any modules you might require:
module load kallisto/0.45.0
#
# Write your commands in the next line
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417885 ../trimmed/SRR6417885* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417886 ../trimmed/SRR6417886* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417887 ../trimmed/SRR6417887* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417888 ../trimmed/SRR6417888* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417889 ../trimmed/SRR6417889* 
kallisto quant -i ../../resources/index_kallisto45_gencode_human_v38 -o ./SRR6417890 ../trimmed/SRR6417890* 
```

```{bash}
qsub quantification.sge
```

### Data integration

Using tximport

```{bash}
screen -S txi
cd /mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/kallisto
module load r/4.0.2
R
```


```{r}
library(tximport)
library(tidyverse)
files <- file.path("/mnt/Timina/bioinfoII/ssalazar/rna-seq/tarea/out/kallisto", list.dirs("."), "abundance.h5")
names(files) <- str_extract(files,"SRR\\d+")
files <- files[c(2, 3, 4, 5, 6, 7)]
```

Transcripts need to be associated with gene IDs for gene-level summarization
For kallisto, the files only provide the transcript ID.
We first make a data.frame called tx2gene with two columns: 1) transcript ID and 2) gene ID.

```{r}
# Loading tables with the info

# transcript id - gene id correspondence
tx2gene <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.v38.basic.pc.transcripts.enstid_ensgid-nover.csv",stringsAsFactors = F)
# transcript id - gene name correspondence
tx2genename <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.v38.basic.pc.transcripts.enstid_genename-nover.csv",stringsAsFactors = F)

```


Runing tximport to summarize transcript-level abundance estimates for transcript- and gene-level analysis.

```{r}
txi.kallisto <- tximport(files, type = "kallisto", tx2gene = tx2gene, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)
txi.kallisto.name <- tximport(files, type = "kallisto", tx2gene = tx2genename, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)

```

```{r}
names(txi.kallisto)
```

[1] "abundance"           "counts"              "length"             
[4] "countsFromAbundance"

```{r}
head(txi.kallisto$counts)
```

                 SRR6417885  SRR6417886 SRR6417887 SRR6417888 SRR6417889
ENSG000000000035  48.856706 36.12039819  24.279581  39.413793  37.693556
ENSG00000000005    0.000000  0.00000000   0.000000   0.000000   0.000000
ENSG000000004194   0.000000  1.81937909   0.000000   5.729418  22.804753
ENSG000000004574   6.663615  0.01752127   7.742207   3.390685   8.781861
ENSG000000004607   9.369975 11.00000000  13.829627   8.001456  16.942265
ENSG000000009383   0.000000  0.00000000   0.000000   0.000000   0.000000
                 SRR6417890
ENSG000000000035  46.855202
ENSG00000000005    0.000000
ENSG000000004194  23.005803
ENSG000000004574   0.000000
ENSG000000004607   6.367758
ENSG000000009383   0.000000

### Normalization

## References

https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html

http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf

http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/latest_release/_README.TXT

https://pachterlab.github.io/kallisto/manual
